k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -962,6 +973,25 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return true;
     }
 
+    private void goBack() {
+        if (changesMadeSinceLastSave && draftIsNotEmpty()) {
```

Comment:
This code duplicates a whole bunch of stuff in onBackPressed. That's unhelpful as if there's a need to change it it'll have to be done in multiple places.